sound/soc/sof/sof-pci-dev.c

```diff
@@ -59,6 +59,13 @@ static const struct dmi_system_id sof_tplg_table[] = {
 		},
 		.driver_data = "sof-adl-rt5682-ssp0-max98373-ssp2.tplg",
 	},
+	{
+		.callback = sof_tplg_cb,
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_NAME, "Felwinter"),
```

Comment:
Humm, I am not sure I get this. We discussed with Google that they would use the DMI_OEM_STRING.
It'd be a bit odd to use a PRODUCT_NAME only

@cujomalainey @bzhg need your help here.

Reply:
Yeap I saw some dmi quirks are using oem string. Is there coding convention or naming rule?

Reply:
Even then some OEMs are not happy that we are using OEM string. @plbossart  Would it be worth allowing the AP FW to express a custom topology file name?

Reply:
well it's a bit late @cujomalainey since we've already blown the DMI_OEM_STRING to be an identifier that is translated to a topology file by the kernel. If you change the semantic of that string to be a topology file that would be problematic...

I am not sure what other strings are left for the platform firmware to populate?

Reply:
I was thinking more like passing it through platform data if possible to the SOF driver. That being said I am not an expert in coreboot, so I have no idea if that is possible

Reply:
same here. It took us a while to converge on DMI_OEM_STRING and it's a bit disappointing that this isn't good enough - not sure what OEMs find objectionable, is this really problematic or a case of work is hard and life is unfair?

Reply:
The OEM complains their tools don't like the fact their name is not in the string. Shall I setup a sync between us and a coreboot member on our team to see if this is feasible?

Reply:
well, if it's just a naming issue that's a problem between the Coreboot team and the OEM
We only need a string to identify a topology, that's a different level.

Reply:
The problem is if they change it, it breaks SOF as it stops checking the community folder for the SOF FW. As for topology names, does pdata sound reasonable?

Reply:
Spoke with a coreboot member on our side, they had no issue with attaching some platform data to the driver and using that to override the topology name. The benefit also being that this won't affect udev since udev can only control path, not name.

Reply:
not sure if we can do it since latest intel platform is using pci enumeration instead of acpi or dt. maybe dmi is still a good place to save those information?

Reply:
not sure if we can do it since latest intel platform is using pci enumeration instead of acpi or dt. maybe dmi is still a good place to save those information?

Reply:
Agree with @brentlu, I have no idea where we would get the platform data from.

Reply:
Isn't [this](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/pci/pci-acpi.c#L1348) an example of a device property being pulled for a PCI device? We could add a string field for the topology name

Reply:
possibly. Be mindful of possible pushback to document such platform firmware properties. It's required for DeviceTree, I don't have a clue if this is acceptable without some written description of this de-facto ABI.

We would probably need some name-spacing anyways to make sure this is limited to Google/Chromebooks.

Reply:
Why would we need the name spacing? You can add a "if property exists" check and then only apply it on platforms that want to use it

Reply:
Perhaps the _DSM (Device Specific Method) is applicable here? This is generic, any platform that may wish to expose this information via firmware (after all, this is platform-specific and not necessarily something the kernel can physically probe for, hence firmware must pass this knowledge to the kernel).

Reply:
> Why would we need the name spacing? You can add a "if property exists" check and then only apply it on platforms that want to use it

look at Device Tree, all properties start with e.g. 'google,topology-name'. the example your referred to is problematic in that this could be abused big time.

Reply:
> Perhaps the _DSM (Device Specific Method) is applicable here? This is generic, any platform that may wish to expose this information via firmware (after all, this is platform-specific and not necessarily something the kernel can physically probe for, hence firmware must pass this knowledge to the kernel).

it's the same, when you get information from platform firmware there is an expectation that the data that you retrieve is documented somehow. Exhibit A for this is the NHLT specification.

Reply:
For context, @twawrzynczak is a chromie coreboot guy who I have asked for expertise on this issue.

I am not sure how this would be abused? I thought coreboot was a trusted source of information

Reply:
Documentation though, yes i agree that would be needed

Reply:
> I am not sure how this would be abused? I thought coreboot was a trusted source of information

Abused by others...

Reply:
Can you explain a little more how you think this is open to abuse? If somebody modifies their firmware to produce the wrong topology (and of course, if you're modifying your firmware, all bets are off if you don't know what you're doing), then of course you can't expect things to still work.

Please allow me to quote from the ACPI spec 6.4 here from section 9.1.1 about _DSM:

```
Successive revisions of Function Arguments must be backward compatible with earlier revisions. New UUIDs may
also be created by OEMs and IHVs for custom devices and other interface or device governing bodies (e.g. the PCI
SIG), as long as the UUID is different from other published UUIDs. Only the issuer of a UUID can authorize a new
Function Index, Revision ID or Function Argument for that UUID.
```

Sometimes _DSM is the best way to inform the kernel of something before there is a standard way to do things in ACPI.

I can think of a few different ways to solve this off the top of my head:
1) Create an ACPI hierarchy of devices (that don't exist in the spec yet) and use properties on these to determine what the audio topology looks like
2) Use a topology string that the kernel can parse in a standard format that can be used across generations

This is to avoid having to reinvent the wheel everytime when the ACPI spec is slower to move than real hardware

Reply:
If somebody modifies their firmware to point to a non-existent topology file on the file system, then we will get a bug report that 'audio is broken' and we will have to add a quirk in the kernel.  Been there, done that.

It's not about the mechanism to retrieve the information, it's about making sure that the bar is high-enough for compliance to a minimal documentation.

Reply:
> Isn't [this](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/pci/pci-acpi.c#L1348) an example of a device property being pulled for a PCI device? We could add a string field for the topology name

Seems the code is for pci root port, it's enumerated from acpi but the pci device should be enumerated by pci itself. It's my understanding thought I'm not pci expert.